src/darwin/Framework/CHIP/templates/helper.js

```diff
@@ -145,6 +145,40 @@ function commandHasRequiredField(command)
   return command.arguments.some(arg => !arg.isOptional);
 }
 
+function enumName(clusterName, enumLabel)
+{
+  clusterName = appHelper.asUpperCamelCase(clusterName);
+  enumLabel   = appHelper.asUpperCamelCase(enumLabel);
+  // Some enum names have one or more copies of the cluster name at the
+  // beginning.
+  while (enumLabel.startsWith(clusterName)) {
+    enumLabel = enumLabel.replace(new RegExp("^" + clusterName), "");
```

Comment:
would `enumLabel = enumLabel.substring(clusterName.length)` work and not use regexes? I would prefer string operations to regex especially if the expression has a custom string input (even if special characters are very very unlikely in cluster names)

Reply:
Yes, that would work.  Can do.